src/bindings/python/src/openvino/runtime/ie_api.py

```diff
@@ -22,23 +22,37 @@ def tensor_from_file(path: str) -> Tensor:
     return Tensor(np.fromfile(path, dtype=np.uint8))
 
 
-def normalize_inputs(inputs: Union[dict, list], py_types: dict) -> dict:
-    """Normalize a dictionary of inputs to Tensors."""
-    if isinstance(inputs, list):
-        inputs = {index: input for index, input in enumerate(inputs)}
+def convert_dict_items(inputs: dict, py_types: dict) -> dict:
+    """Helper function converting dictionary items to Tensors."""
+    # Create new temporary dictionary.
+    # new_inputs will be used to transfer data to inference calls,
+    # ensuring that original inputs are not overwritten with Tensors.
+    new_inputs = {}
     for k, val in inputs.items():
-        if not isinstance(k, (str, int)):
-            raise TypeError("Incompatible key type for tensor named: {}".format(k))
+        if not isinstance(k, (str, int, Output, ConstOutput)):
+            raise TypeError("Incompatible key type for tensor: {}".format(k))
         try:
             ov_type = py_types[k]
         except KeyError:
-            raise KeyError("Port for tensor named {} was not found!".format(k))
-        inputs[k] = (
+            raise KeyError("Port for tensor {} was not found!".format(k))
+        # Convert numpy arrays or copy Tensors
+        new_inputs[k] = (
             val
             if isinstance(val, Tensor)
             else Tensor(np.array(val, get_dtype(ov_type)))
```

Comment:
BTW, why do we copy numpy array here? If we want to copy all the data that user passes in infer functions from performance side better is to perform copy to already allocated InferRequest's tensors.